engine/common.ftl

```diff
@@ -1,5 +1,59 @@
 [#ftl]
```

Comment:
If generic, these functions should be in base.ftl rather than common.ftl.

Reply:
Done